### DISCUSSION TITLE
Way to delete guild-only (or global) slash commands?

### DISCUSSION THREAD

**Discussion by brotherelric** [discord-py-ui/discord-ui#109] [Q&A] [answered]
Title.

**Comment by kvsxxx** [comment, marked as answer]
in the current version there is no real way to delete commands manually other than use the `delete_unused` keyword in the `UI` instance initialization and remove the command

```py
ui = UI(slash_options={"delete_unused": True}) 
```
This will delete all commands in the api that are not registered in the code

For example you have 3 commands, command `A`, `B`, `C`

```py
@ui.slash.command("A")
async def A(ctx):
    ...
@ui.slash.command("B")
async def B(ctx):
    ...
@ui.slash.command("C")
async def C(ctx):
    ...
```

And you want to have command `B` deleted, you just remove the "declaration" for command `B`

```py
@ui.slash.command("A")
async def A(ctx):
    ...

# command B will be deleted when commands are synced

@ui.slash.command("C")
async def C(ctx):
    ...
```

> Note that this will remove **every** command that is registered in the api and was not declared in your code

- - -

in the preview version it's easy, you use `await yourcommand.delete()`

```py
@ui.slash.command(...)
async def my_command(...):
    ...

await my_command.delete()
```

Or you use the method mentioned above (with `delete_unused` keyword)

**Reply by brotherelric** [reply]
This is what I wanted. Thanks!

**Comment by girishladha** [comment]
Yaa !! Actually I was thinking that if we could design a program which helps us to delete msg manually with out using of that old version codes as delete_unsend ...so it might be quite easy to use instead of the old technique..🙄
